extension/src/extensions/python.ts

```diff
@@ -11,15 +11,24 @@ interface Settings {
   }
 }
 
-type EnvironmentVariables = { readonly [key: string]: string | undefined }
+type EnvironmentVariables = { readonly [key: string]: undefined }
 type EnvironmentVariablesChangeEvent = {
   readonly env: EnvironmentVariables
 }
 
+interface Environment {
+  id: string
+  environment?: {
+    type: string
+  }
+}
+
 export interface VscodePython {
   ready: Thenable<void>
   settings: Settings
   environments: {
+    known: Environment[]
+    getActiveEnvironmentPath: () => { id: string }
```

Comment:
[I] If we are using this we also need to use `onDidChangeActiveEnvironmentPath` so that updates are taken into account... unless an update will be triggered in a different way.

[Q] Can we replace our other use of the API with this now?

Reply:
> [I] If we are using this we also need to use onDidChangeActiveEnvironmentPath so that updates are taken into account... unless an update will be triggered in a different way.

Updates are triggered with `onDidChangePythonExecutionDetails` since that runs when the environment path changes. 

> [Q] Can we replace our other use of the API with this now?

Apologies, I'm not sure what you mean. What other use of the API could be replaced? Looking at the API, they all seem to be doing different things to me 🤔

Reply:
Going to merge this, but happy to update the use of the API in a followup!

Reply:
We use `getExecutionDetails` to get the Python executable for the active environment. Isn't that returned by `getActiveEnvironmentPath`?

```
export type EnvironmentPath = {
    /**
     * The ID of the environment.
     */
    readonly id: string;
    /**
     * Path to environment folder or path to python executable that uniquely identifies an environment. Environments
     * lacking a python executable are identified by environment folder paths, whereas other envs can be identified
     * using python executable path.
     */
    readonly path: string;
};
```

☝🏻 maybe we can't rely on this as it could return a folder and not an executable. Nevermind.